app/controllers/targets_controller.rb

```diff
@@ -35,6 +36,14 @@ def destroy
     redirect_to new_target_path
   end
 
+  def matches
+    @matches = []
+    current_user.targets.all.each do |t|
```

Comment:
you can just do `@matches = current_user.targets.map { |t| t.matches }` and then keep the line below to flatten the array. This way you don't have to initialize the empty array